src/OpenTelemetry.Api/Trace/TelemetrySpanNew.cs

```diff
@@ -0,0 +1,187 @@
+﻿// <copyright file="TelemetrySpanNew.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// <para>Span represents the execution of the certain span of code or span of time between two events which is part of
+    /// a distributed trace and has result of execution, context of execution and other properties.</para>
+    /// </summary>
+    public class TelemetrySpanNew : IDisposable
+    {
+        internal static TelemetrySpanNew NoOpInstance = new TelemetrySpanNew(null);
+        private Activity activity;
+
+        internal TelemetrySpanNew(Activity activity)
+        {
+            this.activity = activity;
+        }
+
+        /// <summary>
+        /// Gets the span context.
+        /// </summary>
+        public SpanContextNew Context
+        {
+            get
+            {
+                if (this.activity == null)
+                {
+                    return default;
+                }
+                else
+                {
+                    return new SpanContextNew(this.activity.Context);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether this span will be recorded.
+        /// </summary>
+        public bool IsRecording
+        {
+            get
+            {
+                return (this.activity == null) ? false : this.activity.IsAllDataRequested;
+            }
+        }
+
+        /// <summary>
+        /// Sets the status of the span execution.
+        /// </summary>
+        public Status Status
+        {
+            set
```

Comment:
No getter for Status?

Reply:
TelemetrySpan(New) is not expected to have any Getters except for Context and IsRecording flag.
In Activity, there are no restrictions like this enforced.

OTel guideline is that the API should, in general, only allow setters. And in Exporter, only Getter.